internal/featuretests/v3/featuretests.go

```diff
@@ -66,7 +66,15 @@ const (
 	secretType   = resource.SecretType
 )
 
-func setup(t *testing.T, opts ...interface{}) (cache.ResourceEventHandler, *Contour, func()) {
+// ResourceEventHandlerWrapper wraps the ResourceEventHandler interface from client-go/tools/cache.
+// The OnAdd function has a boolean parameter that we do not need for testing.
+type ResourceEventHandlerWrapper interface {
```

Comment:
ah unfortunate client-go doesn't have this interface, though they would since they do have this: https://github.com/kubernetes/client-go/blob/4cb373f7caddf872f8dd0e0adbed3d0fe7c6aaf7/tools/cache/controller.go#L234-L238

Reply:
@sunjayBhatia I wonder if this kind of abrupt non-backwards compatible change (the addition of `isInInitialList bool` parameter) in a hugely popular package could be a sign that client-go developers did not expect people to depend on `ResourceEventHandler` interface and redefine their own `OnAdd` etc methods directly? It seemed strange decision to break clients just because of new parameter like that.  Maybe they expected users only to instantiate `ResourceEventHandlerFuncs` struct which you point out, and not define their own like Contour does.

Reply:
I think the docs on the `*Funcs` structs imply theyre for usage when you don't need to implement all the methods, so I would expect the authors to know people will rely on the interface itself

there is also `*DetailedFuncs` here: https://github.com/kubernetes/client-go/blob/4cb373f7caddf872f8dd0e0adbed3d0fe7c6aaf7/tools/cache/controller.go#L264

regardless, unfortunate there was such a change w/o a major version revision